src/common/operator/cast_operators.cpp

```diff
@@ -900,8 +899,9 @@ template <> string_t CastToBlob::Operation(string_t input, Vector &vector) {
 	string_t result;
 	// Check by a hex string
 	if(input_size >= 2 && input_data[0] == '\\' && input_data[1] == 'x') {
-		auto aux = CastFromBlob::FromHexToBytes(string(input_data));
-		result = StringVector::AddBlob(vector, string_t(aux));
+		auto output = StringVector::EmptyString(vector, (input_size - 2) / 2);
+		CastFromBlob::FromHexToBytes(input, output);
+		result = StringVector::AddBlob(vector, output);
```

Comment:
This is not required; if you use StringVector::EmptyString, the returned object is already added to the string heap. Performing the `StringVector::AddBlob` here will initiate a copy. This can simply be `result = output;` and it should still work.